sklearn/pipeline.py

```diff
@@ -26,6 +26,33 @@
 __all__ = ['Pipeline', 'FeatureUnion', 'make_pipeline', 'make_union']
 
 
+# weight and fit_params are not used but it allows _fit_one_transformer,
+# _transform_one and _fit_transform_one to have the same signature to
+# factorize the code in ColumnTransformer
+def _fit_one_transformer(transformer, X, y, weight=None, **fit_params):
+    return transformer.fit(X, y)
+
+
+def _apply_weight(Xt, weight=None):
+    # if we have a weight for this transformer, multiply output
+    if weight is None:
+        return Xt
+    return Xt * weight
```

Comment:
I'm not sure there is really a need to factorize the code into this additional function. Here we only have 2 repetitions with little code, and a rule of thumb (source forgotten) is to factorize after 3..

Reply:
looking at the code base right now I'd agree with you, but it won't hurt to keep it as a static method (and I use this as well :grimacing: )

Reply:
Yes, but if it's only used  in `_transform_one` and `_fit_transform_one` you could always overwrite those two functions and define `_apply_weight` as need, can't you? (or copy this function into your package).

Reply:
I could, but what would be the benefit of doing so?